src/pages/institution/institution-service.ts

```diff
@@ -0,0 +1,150 @@
+import { HttpClient } from "@angular/common/http";
+import { Injectable } from "@angular/core";
+import { UtilsProvider } from "../../providers/utils/utils";
+import { AppConfigs } from "../../providers/appConfig";
+import { UpdateLocalSchoolDataProvider } from "../../providers/update-local-school-data/update-local-school-data";
+import { ApiProvider } from "../../providers/api/api";
+import { LocalStorageProvider } from "../../providers/local-storage/local-storage";
+import { storageKeys } from "../../providers/storageKeys";
+
+/*
+  Generated class for the InstitutionServiceProvider provider.
+
+  See https://angular.io/guide/dependency-injection for more info on providers
+  and Angular DI.
+*/
+@Injectable()
+export class InstitutionServiceProvider {
+  constructor(
+    public http: HttpClient,
+    public utils: UtilsProvider,
+    private apiService: ApiProvider,
+    private ulsdp: UpdateLocalSchoolDataProvider,
+    private localStorage: LocalStorageProvider
+  ) {
+    console.log("Hello InstitutionServiceProvider Provider");
+    console.log("Hello InstitutionServiceProvider Provider");
+  }
+
+  getInstituionFromStorage() {
+    // this.utils.startLoader();
+    return this.localStorage
+      .getLocalStorage(storageKeys.institutionFlowList)
+      .then((data) => {
+        if (data) {
+          // this.migrationFuntion(data);
+          return data;
+        } else {
+          return this.getInstitutionsflowApi();
+        }
+      })
+      .catch((error) => {
+        return this.getInstitutionsflowApi();
+      });
+  }
+
+  getInstitutionsflowApi() {
+    return new Promise((resolve, reject) => {
+      const url = AppConfigs.institutionsFlow.institutions;
+
+      this.apiService.httpGet(
+        url,
+        (successData) => {
+          this.localStorage.setLocalStorage(
+            storageKeys.institutionFlowList,
+            successData.result
+          );
+          resolve(successData.result);
+        },
+        (error) => {
+          reject();
+        }
+      );
+    });
+  }
+
+  getAssessmentDetails(event, institutionList) {
```

Comment:
This is same in programs flow also right?

Reply:
here there is difference in getting the ids of program solution and entity 
eg
in program flow entityId is : 
let entityId =
        programs[programIndex].solutions[assessmentIndex].entities[entityIndex]
          ._id;
while in institution flow it is     
 let entityId = institutionList.entities[entityType][entityIndex]._id;